model/task/task.go

```diff
@@ -1442,6 +1443,40 @@ func (t *Task) SetStepbackDepth(stepbackDepth int) error {
 		})
 }
 
+// SetLogServiceVersion sets the log service version used to write logs for the
+// task.
+func (t *Task) SetLogServiceVersion(ctx context.Context, env evergreen.Environment, version int) error {
+	if t.DisplayOnly {
+		return errors.New("cannot set log service version on a display task")
+	}
+	if t.LogServiceVersion != nil {
+		return errors.New("log service version already set")
+	}
+
+	res, err := env.DB().Collection(Collection).UpdateByID(ctx, t.Id, []bson.M{
+		{
+			"$set": bson.M{LogServiceVersionKey: bson.M{
+				"$ifNull": bson.A{
+					"$" + LogServiceVersionKey,
+					version,
+				}},
+			},
+		},
+	})
```

Comment:
Why is it preferable to include the Null check in the update instead of in the filter?
Like why couldn't something like this work?
```go
res, err := env.DB().Collection(Collection).UpdateOne(ctx,
    bson.M{IdKey: t.Id, LogServiceVersionKey: nil},
    bson.M{"$set": bson.M{LogServiceVersionKey: version}},
)
```

Reply:
It could, but then we wouldn't know whether there is a programmatic error (the task ID should should exist does not exist) or the field is NULL.